src/amdgpu_libdrm.cpp

```diff
@@ -0,0 +1,85 @@
+#include <deque>
+#include <mutex>
+#include <thread>
+#include <chrono>
+#include <libdrm/amdgpu.h>
```

Comment:
`s|libdrm/||` - the `dependency` suggestion above will ensure the correct folder is passed during compilation.

Reply:
I think that `libdrm/` can't be omitted here, even with `dependency` instead of `cpp.find_library`

Reply:
If that doesn't work either the dependency is not added to the correct place in meson.build or the distro is horribly broken.

Reply:
I might need to add `libdrm` as a separate dependency for this to work. I'll try it shortly

Reply:
It seems that the problem is that it includes the local `./amdgpu.h` instead of the global `/usr/include/libdrm/amdgpu.h` and I'm not sure what to do about that

Reply:
Not sure what you mean here - there is no `amdgpu.h` in tree to get confused with. Unless there is a local one in your git checkout?

Reply:
[There absolutely is an amdgpu.h in tree](https://github.com/flightlessmango/MangoHud/blob/master/src/amdgpu.h)

Reply:
`src/` seems to be on the include path for some reason

Reply:
> src/ seems to be on the include path for some reason

This should make no different what-so-ever. What is the exact `#include` directive you're using and the error log being produced?

Reply:
With `#include <amdgpu.h>` it outputs
```
../src/amdgpu_libdrm.cpp:39:1: error: ‘amdgpu_device_handle’ does not name a type
   39 | amdgpu_device_handle amdgpu_handle;
      | ^~~~~~~~~~~~~~~~~~~~
../src/amdgpu_libdrm.cpp: In function ‘void libdrm_do_sample(libdrm_sample*)’:
../src/amdgpu_libdrm.cpp:43:30: error: ‘amdgpu_handle’ was not declared in this scope
   43 |     amdgpu_read_mm_registers(amdgpu_handle, LIBDRM_GRBM_STATUS / 4, 1, 0xffffffff, 0, &registers);
      |                              ^~~~~~~~~~~~~
../src/amdgpu_libdrm.cpp:43:5: error: ‘amdgpu_read_mm_registers’ was not declared in this scope
   43 |     amdgpu_read_mm_registers(amdgpu_handle, LIBDRM_GRBM_STATUS / 4, 1, 0xffffffff, 0, &registers);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
../src/amdgpu_libdrm.cpp: In function ‘int libdrm_initialize()’:
../src/amdgpu_libdrm.cpp:103:69: error: ‘amdgpu_handle’ was not declared in this scope
  103 |     if (amdgpu_device_initialize(fd, &libdrm_major, &libdrm_minor, &amdgpu_handle)) {
      |                                                                     ^~~~~~~~~~~~~
../src/amdgpu_libdrm.cpp:103:9: error: ‘amdgpu_device_initialize’ was not declared in this scope
  103 |     if (amdgpu_device_initialize(fd, &libdrm_major, &libdrm_minor, &amdgpu_handle)) {
      |         ^~~~~~~~~~~~~~~~~~~~~~~~
```

If I add an `#include <libdrm/amdgpu.h>` to the local `./amdgpu.h` it compiles. IMO the only possible explanation for this is that the compiler includes the local one, even with the angle brackets

Reply:
```
c++ -Isrc/libMangoHud.a.p -Isrc -I../src -I../include -Isubprojects/imgui-1.81 -I../subprojects/imgui-1.81 ... -I/usr/include/libdrm ... -c ../src/amdgpu_libdrm.cpp
../src/amdgpu_libdrm.cpp:39:1: error: ‘amdgpu_device_handle’ does not name a type
```

From the above - meson is inserting `-Isrc -I../src` at the very start, even though there is no `include_directories` directives instructing it to. Perhaps something in our meson.build is forcing it or maybe not. In the latter case this is fairly concerning.

As a quick and dirty approach I would simply rename the in-tree file - as a separate commit.

Reply:
Meson seems to always -I the directory in which the source files reside. Just tested it on an almost empty project